src/game_initialization/multiplayer.cpp

```diff
@@ -572,18 +580,27 @@ bool mp_manager::enter_lobby_mode()
 	return true;
 }
 
-void mp_manager::enter_create_mode()
+void mp_manager::enter_create_mode(utils::optional<std::string> preset_scenario)
 {
 	DBG_MP << "entering create mode";
 
-	if(gui2::dialogs::mp_create_game::execute(state, connection == nullptr)) {
-		enter_staging_mode();
+	if(preset_scenario) {
+		for(const config& game : game_config_manager::get()->game_config().mandatory_child("game_presets").child_range("game")) {
+			if(game["scenario"].str() == preset_scenario.value()) {
+				// NOLINTNEXTLINE(performance-unnecessary-value-param)
```

Comment:
it's imo better to make the copy explicit rather than making it implicit it in the funciton call, also the line above can probalby be simplified by using config::find_child()

Maybe we could even use something like
```
auto game = game_config_manager::get()->game_config().mandatory_child("game_presets")
	.find_child("game", "scenario",  preset_scenario.value())
	// make a copy because quick_mp_setup reloads the game config.
	.and_then([](const confg& cfg) -> std::optional<config>{ return cfg; });
if(game) {
  gui2::dialogs::mp_create_game::quick_mp_setup(state, *game);
 ...
}
```

The advantage of such a construction would be that there at no point exists a reference variable pointing to a destructed object (just an idea tho, i'm not trying to establish this style of coding).

Reply:
Would there be any benefit to doing that vs just adding something like `config g = game`?

Reply:
hmm not sure to what you are comparing it here, since the examplecode combined multiple suggested changes. so
- The benefit of using find_child is just simplifiying the code (this is exactly the usecase that that function was made for)
- The benefit of and_then is as i said, that there at no point a reference of a destroyed variable (here game) being in scope (meaning  you cannot accidentally write `somefunction(game)` after `quick_mp_setup` (where game is a pointer to the game_config) because no such variable exists), but its mostly a matter of coding preferences imo.

Reply:
`class "optional_config_impl<const config>" has no member "and_then"`, so it doesn't look like that's currently supported anyway.